teams.md/docs/typescript/migrations/v1/README.md

```diff
@@ -3,8 +3,614 @@ sidebar_position: 2
+server.use(restify.plugins.bodyParser());
+
+server.listen(process.env.port || process.env.PORT || 3978, () => {
+    console.log(`\n${server.name} listening to ${server.url}`);
+    console.log('\nTo test your bot in Teams, sideload the app manifest.json within Teams Apps.');
+});
+
+// Define storage and application
+const app = new Application<ApplicationTurnState>({
+    storage: new MemoryStorage()
+});
+
+// Listen for incoming server requests.
+server.post('/api/messages', async (req, res) => {
+    // Route received a request to adapter for processing
+    await adapter.process(req, res, async (context) => {
+        // Dispatch to application for routing
+        await app.run(context);
+    });
+});
+```
+
+Here is the comparable code in Teams AI v2:
+
+```ts
+import { App } from '@microsoft/teams.apps';
+import { LocalStorage } from '@microsoft/teams.common/storage';
+
+// Define app
+const app = new App({
+  clientId: process.env.ENTRA_APP_CLIENT_ID!,
+  clientSecret: process.env.ENTRA_APP_CLIENT_SECRET!,
+  tenantId: process.env.ENTRA_TENANT_ID!,
+});
+
+// Optionally create local storage
+const storage = new LocalStorage();
+
+// Listen for errors
+app.event('error', async ({ error, activity }) => {
+  console.error('Error event received:', error);
+  if (activity) {
+    await app.send(activity.conversation.id, 'An error occurred while processing your message.');
+  }
+});
+
+// App by default will start a local server with a route for /api/messages
+// If you want to reuse your restify or other server,
+// create a custom `HttpPlugin` and pass into new App({ plugins: [plugin] })
+(async () => {
+  // starts the server
+  await app.start();
+})();
+```
+
+## Migrate activity handlers
+
+Both Teams AI v1 and v2 are built atop incoming `Activity` requests, which trigger handlers in your code when specific type of activities are received. The syntax for how you register different types of `Activity` handlers differs between the v1 and v2 versions of our SDK.
+
+### Message handlers
+
+Here is an example of a message handler in Teams AI v1:
+
+```ts
+// will be triggered when user sends "/hello" or "@yourbot /hello"
+app.message(
+  "/hello",
+  async (context: IConversationContext) => {
+    await context.sendActivity("Hello!");
+  }
+);
+// listen for ANY message to be received
+app.activity(ActivityTypes.Message, async (context: TurnContext) => {
+    // echo back users request
+    await context.sendActivity(`you said: ${context.activity.text}`);
+});
+```
+
+Here is the equivalent code in Teams AI v2:
+
+```ts
+// will be triggered when user sends "/hello" or "@yourbot /hello"
+app.message('/hello', async ({ send }) => {
+  // Teams AI v2 does not automatically send typing indicators
+  await send({ type: 'typing' });
+  await send("Hello!");
+});
+// listen for ANY message to be received
+app.on('message', async ({ send, activity }) => {
+  await send({ type: 'typing' });
+  await send(`you said "${activity.text}"`);
+});
+```
```

Comment:
instead of destructing the context, you could keep it there, so it's easier to see the difference.